code_quality.js

```diff
@@ -0,0 +1,211 @@
+   }
+   Object.keys(filesByExtension).forEach(extension => {
+      const files = filesByExtension[extension];
+      const config = prettierConfig[extension];
+      if (!config) {
+         console.log(
+            "We don't have any configuration for extension " +
+               extension +
+               ', please add it in file prettier_config.json'
+         );
+         return;
+      }
+      files.forEach(file => prettyFile(file, config));
+   });
+};
+
+const eslintFiles = async files => {
+   if (!codeQualityConfig.useEslint) {
+      return {};
+   }
+
+   const formatter = await eslint.loadFormatter('stylish');
+   let error = false;
+   let report = '';
+   for (const file of files) {
+      try {
+         const result = await eslint.lintFiles(file);
+         error = error || result.some(res => res.errorCount > 0);
+         await ESLint.outputFixes(result);
+         const resultText = formatter.format(result);
+         if (!!resultText && resultText.trim().length > 0) {
+            report += resultText;
+         }
+      } catch (exception) {
+         error = true;
+         report += 'Error in eslint file ' + file + ':\n' + exception;
+      }
+   }
+
+   return { error, report };
+};
+
+const preCommit = async () => {
+   const files = await getChangedFiles();
+   if (!files) {
+      console.log("Couldn't find any file that hand been changed");
+      return;
+   }
+   console.log('File to be analyzed before commit:\n' + files.join('\n'));
+   const filesByExtension = groupFilesByExtension(files);
+   prettyFiles(filesByExtension);
+
+   let error = false;
+   let report = '';
+   const extensions = Object.keys(filesByExtension);
+   for (const extension of extensions.filter(
+      extension => extension === 'ts' || extension === 'js'
+   )) {
+      const eslintResult = await eslintFiles(filesByExtension[extension]);
+      error = error || eslintResult.error;
+      if (!!eslintResult.report && eslintResult.report.trim().length > 0) {
+         report += eslintResult.report;
+      }
+   }
+
+   if (error) {
+      console.log('Please fix these errors before to commit:' + report);
+      if (codeQualityConfig.eslintStopCommit) {
+         process.exit(1);
+      }
+   }
+};
+
+const full = async () => {
+   const files = getAllFiles(__dirname);
+   const filesByExtension = groupFilesByExtension(files);
+   prettyFiles(filesByExtension);
+
+   let error = false;
+   let report = '';
+   const extensions = Object.keys(filesByExtension);
+   for (const extension of extensions.filter(
+      extension => extension === 'ts' || extension === 'js'
+   )) {
+      const eslintResult = await eslintFiles(filesByExtension[extension]);
+      error = error || eslintResult.error;
+      if (!!eslintResult.report && eslintResult.report.trim().length > 0) {
+         report += eslintResult.report;
+      }
+   }
+
+   if (error) {
+      console.log('Resume of ESLint analysis:\n' + report);
```

Comment:
The Resume seems to be giving pretty interesting insights. In the rank of thousands though. Probably we need to have a discussion in next meeting.

Reply:
Yes, probably we can write a file with Resume of full es lint analysis (cause the result is pretty... huge)... 
By the way, I would expect that the full analysis should never be used (or just once).... 
But yes, we can discuss it tomorrow morning during the daily

Reply:
👍

Reply:
I changed a bit the code, now eslint full report will be generated and converted in html to file full_eslint_report.html (same directory and it will be git ingored.)